src/server/rest/v1/service/ChargingStationService.ts

```diff
@@ -114,12 +116,40 @@ export default class ChargingStationService {
     if (Utils.objectHasProperty(filteredRequest, 'forceInactive')) {
       chargingStation.forceInactive = filteredRequest.forceInactive;
     }
+    if (Utils.objectHasProperty(filteredRequest, 'manualConfiguration')) {
+      // Check manual config
+      if (!chargingStation.manualConfiguration && filteredRequest.manualConfiguration) {
+        chargingStation.manualConfiguration = filteredRequest.manualConfiguration;
+        delete chargingStation.templateHash;
+        delete chargingStation.templateHashCapabilities;
+        delete chargingStation.templateHashOcppStandard;
+        delete chargingStation.templateHashOcppVendor;
+        delete chargingStation.templateHashTechnical;
+      } else if ((chargingStation.manualConfiguration && !filteredRequest.manualConfiguration) ||
+      (!filteredRequest.manualConfiguration && !(chargingStation.chargePoints?.length > 0))) {
```

Comment:
The first line of the condition checks if the manual configuration is enabled in the charging station and if the request disables it (So the end user does not want to have it manually configured anymore)
--> if yes we have to apply the template again

The second line checks if manual configuration is disabled in the request and if the charging station has the property charge point on the station (stations without a template do not have this property right?)
--> If yes the backend will try to find a template (If no template is found an error is thrown)
This happens when the user tries to disable the manual config on a station without a template

Reply:
> The second line checks if manual configuration is disabled in the request and if the charging station has the property charge point on the station (stations without a template do not have this property right?)

They will in the future, once the UI will be done to edit chargePoints. 
The only boolean to check is the manualConfiguration one.  

> --> If yes the backend will try to find a template (If no template is found an error is thrown)
> This happens when the user tries to disable the manual config on a station without a template

The backend code shall not error out in case of no template found for a charging station model. It just have to log it and maybe notify the dashboard, but no having a template for a charging station is not an error.

Reply:
So do we need a migration task then, which adds the property manual configuration to all the stations without a template?
Or do I have to separate the manual configuration completely from the charging stations without a template?

Reply:
You can have charging station without matching template with no manual configuration flag. It's in fact the default when you connect a charging not known before. 
The manual flag is really relevant when there's a matching a template.
When the charging station have no matching template, the behaviour is the same as with manual configuration but without the flag in the DB.